Civi/Core/LocalizationInitializer.php

```diff
@@ -0,0 +1,160 @@
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+namespace Civi\Core;
+
+use Civi;
+use Civi\Core\Event\SystemInstallEvent;
+
+/**
+ * Class LocalizationInitializer
+ * @package Civi\Core
+ */
+class LocalizationInitializer {
+
+  /**
+   * Load the locale settings based on the installation language
+   *
+   * @param \Civi\Core\Event\SystemInstallEvent $event
+   * @throws \CRM_Core_Exception
+   */
+  public static function initialize(SystemInstallEvent $event) {
+
+    // get the current installation language
+    global $tsLocale;
+    $seedLanguage = $tsLocale;
+    if (!$seedLanguage) {
+      return;
+    }
+
+    // get the corresponding settings file if any
+    $localeDir = \CRM_Core_I18n::getResourceDir();
+    $fileName = $localeDir . $seedLanguage . DIRECTORY_SEPARATOR . 'settings.default.json';
+
+    // initalization
+    $settingsParams = array();
+
+    if (file_exists($fileName)) {
+
+      // load the file and parse it
+      $json = file_get_contents($fileName);
+      $settings = json_decode($json, TRUE);
+
+      if (!empty($settings)) {
+        // get all valid settings
+        $results = civicrm_api3('Setting', 'getfields', array());
+        $validSettings = array_keys($results['values']);
+        // add valid settings to params to send to api
+        foreach ($settings as $setting => $value) {
+          if (in_array($setting, $validSettings)) {
+            $settingsParams[$setting] = $value;
+          }
+
+        }
+
+        // ensure we don't mess with multilingual
+        unset($settingsParams['languageLimit']);
+
+        // support for enabled languages (option group)
+        if (isset($settings['languagesOption']) && count($settings['languagesOption']) > 0) {
+          self::updateLanguages($settings['languagesOption']);
+        }
+
+        // set default currency in currencies_enabled (option group)
+        if (isset($settings['defaultCurrency'])) {
+          self::updateCurrencies(array($settings['defaultCurrency']), $settings['defaultCurrency']);
+        }
+
```

Comment:
2. Oh, no, you can enable language that are not in the list of available preferred languages. But the label is lost in the admin localization setting page - so you get `es_ES` instead of `Spanish` and it gets to the bottom of the list. I have fixed this bug.

3. Except that there is no settings for `currency_enabled`. I have to manually initialize to the same as default language. The logic applies only in the installation process. So i guess i need the code after all.  To avoid duplication of code, I have moved it to `CRM_Admin_Form_Setting_Localization`. We might want to have an simple api call for updating currency_enabled but it's not in the scope of this issue.